docsrc/content/extensions.fsx

```diff
@@ -0,0 +1,295 @@
+(*** hide ***)
+// This block of code is omitted in the generated HTML documentation. Use 
+// it to define helpers that you do not want to show in the documentation.
+#I "../../bin"
+#r @"../../src/FSharpPlus/bin/Release/net45/FSharpPlus.dll"
+
+(**
+Extensions
+=======================
+Extensions are what you probably expect: helper functions for existing types.
+ 
+They are defined as modules with the same name as the types they operate on
+under the FSharpPlus namespace, so can be accessed via:
+*)
+open FSharpPlus
+
+(**
+Some functions are common across traversable types such as `intercalate` on
+List, Array and Seq, and others are common across wrapping containers,
+such as `map`, `bind` and `apply` on List, Array, and Seq, but also Option and Result.
+
+Construction:
+=============
+This is added and is already defined for Seq, Array and List:
+
+ * Enumerator.singleton - construct a container with the given value inside it
+
+To construct MonadError instances (Result or Choice) you can use result/throw:
+
+ * result - construct with the given value (as Ok or Choice1Of2)
+ * throw - construct an error from the given value (as Error or Choice2of2)
+
+It's also possible to construct by wrapping exception producing functions:
+
+ * protect - protect a function in try/catch, returning a wrapped value
+
+    // throws "ArgumentException: The input sequence was empty."
+    let expectedSingleItem = List.exactlyOne []
+
+    // returns a Result.Error holding the exception as its value:
+    let expectedSingleItem = Result.protect List.exactlyOne []
+
+    // ...or like typical try prefixed functions, treat exception as None
+    let expectedSingleItem = Option.protect List.exactlyOne []
+
+    // which might look like this:
+    let inline tryExactlyOne xs = Option.protect List.exactlyOne xs
+
+Deconstruction (unwrapping):
+============================
+Some extensions on Result are designed to behave like Option:
+
+ * Result.get - unwraps the value when it is an 'ok, otherwise throws an exception
+ * Result.defaultValue - return the 'ok value if present, otherwise the default value
+ * Result.defaultWith - return the 'ok value uif present, otherwise apply the given function
+   to the 'error value
+
+To deconstruct MonadError instances (Result or Choice) use:
+
+ * either - unwraps the result/choice by applying the given `ok` or `err` function as appropriate
+
+Note that there is also the generic `either` operator function that works
+exactly the same as `Result.either`.
+
+Also, see the generic function [`option`](reference/fsharpplus-operators.html) that
+unwraps an Option in a similar way to `either`.
+
+On Foldables (collections)
```

Comment:
Note that foldable and collections are actually two different abstractions although they are related.
A collection in some sense is like the opposite to a foldable.

Reply:
Hmmm, in what sense do you mean a collection is opposite?

Reply:
Well, a foldable is something that can reduce to a single value. And a Collection is something that can unfold to many values.

That's why we use toSeq for Foldable and OfSeq for collections.

It's not very clear this explanation as it's not really clear the distinction, specially of the latter, so just take as a general guide.

Reply:
I have never thought of a sequence like an unfoldable, but of course it is :)

Thanks.